services/opentsdb/service.go

```diff
@@ -408,20 +455,21 @@ func (s *Service) serveHTTP() {
 
 // processBatches continually drains the given batcher and writes the batches to the database.
 func (s *Service) processBatches(batcher *tsdb.PointBatcher) {
-	defer s.wg.Done()
 	for {
 		select {
+		case <-s.done:
+			return
 		case batch := <-batcher.Out():
+			// Will attempt to create database if not yet created.
+			s.createInternalStorage()
```

Comment:
But then it continues on to try the write which will fail as well and tie up resources.  Seems like we should abort the write here since we already know it will fail.